### PR TITLE
OHRI-1995 MDR TB Enrolment table under TB program management menu

### PR DESCRIPTION
**Pull request by ODORA0**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary
Aside from creating the table list, we needed to make sure that the user cannot add a new form in enter mode `ADD+`  if there is no Treatment Outcome added. The approach used here is fetching the latest encounter of the patient and check if there is an observation captured for the treatment outcome concept then hide the add button if it is truthy.
## Screenshots
image
image

### CODE REVIEW

**Comment by pirupius** [comment]
@ODORA0 Is this ready for review? I see u still have it on draft and u haven't updated the description but you have assigned the ticket

**Comment by ODORA0** [comment]
@pirupius No it is not, the form wasn't yet ready.

File: packages/esm-tb-app/src/config-schema.ts [hunk 1]
```diff
@@ -37,15 +37,20 @@ export const configSchema = {
       monthOfTreatment: '1418AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
       adherenceAssessment: '164075AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
       nextAppointmentDate: '5096AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+      DateEnrolledToTB: '161552AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+      TreatmentId: '164415AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+      TreatmentStartDate: '164416AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+      Drregimen: '16fd7307-0b26-4c8b-afa3-8362baff4042',
```
Comment (pirupius) [inline packages/esm-tb-app/src/config-schema.ts:43]: There's already a conept `regimen` with the same concept so i suggest to delete this and use the existing

File: packages/esm-tb-app/src/config-schema.ts [hunk 2]
```diff
@@ -37,15 +37,20 @@ export const configSchema = {
       monthOfTreatment: '1418AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
       adherenceAssessment: '164075AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
       nextAppointmentDate: '5096AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+      DateEnrolledToTB: '161552AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
```
Comment (pirupius) [inline packages/esm-tb-app/src/config-schema.ts:40]: This is the same as `enrollmentDate`

File: packages/esm-tb-app/src/config-schema.ts (hunk 1 above)
Comment (pirupius) [inline packages/esm-tb-app/src/config-schema.ts:43]: There's already a conept `regimen` with the same concept so i suggest to delete this and use the existing

File: packages/esm-tb-app/src/views/program-management/tabs/mdr-tb.component.tsx [hunk 3]
```diff
@@ -1,15 +1,119 @@
...
+    });
+  }, []);
+
+  const columns: EncounterListColumn[] = useMemo(
+    () => [
+      {
+        key: 'enrollmentDate',
+        header: t('enrollmentDate', 'Enrollment Date'),
+        getValue: (encounter) => {
+          return getObsFromEncounter(encounter, config.obsConcepts.DateEnrolledToTB, true);
```
Comment (pirupius) [inline packages/esm-tb-app/src/views/program-management/tabs/mdr-tb.component.tsx:34]: update this

**Review by pirupius — CHANGES_REQUESTED** [review]

**Review by pirupius — APPROVED** [review]

File: packages/esm-tb-app/src/config-schema.ts (hunk 1 above)
Comment (ODORA0) [reply packages/esm-tb-app/src/config-schema.ts]: Done

File: packages/esm-tb-app/src/config-schema.ts (hunk 2 above)
Comment (ODORA0) [reply packages/esm-tb-app/src/config-schema.ts]: Done

File: packages/esm-tb-app/src/config-schema.ts (hunk 1 above)
Comment (ODORA0) [reply packages/esm-tb-app/src/config-schema.ts]: Done

File: packages/esm-tb-app/src/views/program-management/tabs/mdr-tb.component.tsx (hunk 3 above)
Comment (ODORA0) [reply packages/esm-tb-app/src/views/program-management/tabs/mdr-tb.component.tsx]: Done

**Pull request merged**